python_testcontainers/infrahub_testcontainers/container.py

```diff
@@ -298,26 +283,120 @@ def database_restore_backup(self, backup_file: Path) -> None:
+                        "-p",
+                        "admin",
+                        "SHOW DATABASES WHERE name = 'neo4j' AND currentStatus = 'online';",
+                    ],
+                    service_name=service_name,
+                )
+                if stdout:
+                    break
+                time.sleep(5)
+            else:
+                (debug_logs, _, _) = self.exec_in_container(
+                    command=["cat", "logs/debug.log"],
+                    service_name=service_name,
+                )
+                raise Exception(f"Failed to restore database:\n{restore_output}\nDebug logs:\n{debug_logs}")
+
+            old_services = self.services
+            self.services = ["infrahub-server", "task-worker"]
+            self.stop(down=False)
+            try:
+                self.start()
+            except Exception as exc:
+                stdout, stderr = self.get_logs()
+                raise Exception(f"Failed to start docker compose:\nStdout:\n{stdout}\nStderr:\n{stderr}") from exc
+            self.services = old_services
         else:
-            (debug_logs, _, _) = self.exec_in_container(
-                command=["cat", "logs/debug.log"],
+            print("Cluster mode detected")
+            try:
+                self.get_container(service_name=service_name)
+                self.get_container(service_name="database-core2")
+                self.get_container(service_name="database-core3")
+            except ContainerIsNotRunning:
+                self.start_container("database", "database-core2", "database-core3")
+
+            # Waiting for cluster to stabilize...
+            time.sleep(10)
+
+            self.exec_in_container(
+                command=["cypher-shell", "-u", "neo4j", "-p", "admin", "DROP DATABASE neo4j;"],
+                service_name=service_name,
+            )
+
+            self.exec_in_container(
+                command=["rm", "-rf", "/data/databases/neo4j"],
+                service_name=service_name,
+            )
+            self.exec_in_container(
+                command=["rm", "-rf", "/data/transactions/neo4j"],
                 service_name=service_name,
             )
-            raise Exception(f"Failed to restore database:\n{restore_output}\nDebug logs:\n{debug_logs}")
 
-        old_services = self.services
-        self.services = ["infrahub-server", "task-worker"]
-        self.stop(down=False)
-        try:
+            self.exec_in_container(
+                command=[
+                    "neo4j-admin", "database", "restore",
+                    "--from-path", str(self.internal_backup_dir / backup_file.name),
+                    "neo4j",
+                ],
+                service_name=service_name,
+            )
+
+            cmd = self.compose_command_property[:]
+            cmd += ["restart", "database"]
+            self._run_command(cmd=cmd)
+            # Waiting for cluster to stabilize...
+            time.sleep(10)
```

Comment:
Need to replace this with a better check (check the cluster state in a loop with timeout?)
Similar to the check above with:

```
           for _ in range(3):
                (stdout, _, _) = self.exec_in_container(
                    command=[
                        "cypher-shell",
                        "--format",
                        "plain",
                        "-d",
                        "system",
                        "-u",
                        "neo4j",
                        "-p",
                        "admin",
                        "SHOW DATABASES WHERE name = 'neo4j' AND currentStatus = 'online';"
```